LEGO1/mxwavepresenter.cpp

```diff
@@ -1,22 +1,213 @@
 #include "mxwavepresenter.h"
 
 #include "decomp.h"
+#include "mxdssound.h"
+#include "mxomni.h"
+#include "mxsoundmanager.h"
+
+#include <limits.h>
 
 DECOMP_SIZE_ASSERT(MxWavePresenter, 0x6c);
+DECOMP_SIZE_ASSERT(MxWavePresenter::WaveFormat, 0x1c);
+
+// OFFSET: LEGO1 0x1000d640
+MxWavePresenter::~MxWavePresenter()
+{
+	Destroy(TRUE);
+}
+
+// OFFSET: LEGO1 0x1000d6a0
+void MxWavePresenter::Destroy()
+{
+	Destroy(FALSE);
+}
+
+// OFFSET: LEGO1 0x1000d6b0
+undefined MxWavePresenter::VTable0x6c()
+{
+	return m_unk68;
+}
 
 // OFFSET: LEGO1 0x100b1ad0
 void MxWavePresenter::Init()
 {
-	m_unk54 = 0;
-	m_unk58 = 0;
-	m_unk5c = 0;
+	m_waveFormat = NULL;
+	m_dsBuffer = NULL;
+	m_length = 0;
 	m_unk60 = 0;
 	m_unk64 = 0;
-	m_unk65 = 0;
-	m_unk66 = 0;
+	m_unk65 = FALSE;
+	m_unk66 = FALSE;
 	m_unk68 = 0;
 }
 
+// OFFSET: LEGO1 0x100b1af0
+MxResult MxWavePresenter::AddToManager()
+{
+	MxResult result = MxSoundPresenter::AddToManager();
+	Init();
+	return result;
+}
+
+// OFFSET: LEGO1 0x100b1b10
+void MxWavePresenter::Destroy(MxBool p_fromDestructor)
+{
+	if (m_dsBuffer) {
+		m_dsBuffer->Stop();
+		m_dsBuffer->Release();
+	}
+
+	if (m_waveFormat)
+		delete[] ((MxU8*) m_waveFormat);
+
+	Init();
+
+	if (!p_fromDestructor)
+		MxSoundPresenter::Destroy(FALSE);
+}
+
+// OFFSET: LEGO1 0x100b1b60
+MxS8 MxWavePresenter::FUN_100b1b60()
+{
+	DWORD dwCurrentPlayCursor, dwCurrentWriteCursor;
+	MxS8 result = -1;
```

Comment:
```suggestion
	MxS8 result = FAILURE;
```

Reply:
The type is not `MxResult` but `MxS8` instead, so this would not be appropriate.